ui-frontend/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter.structure.tsx

```diff
@@ -164,6 +167,11 @@ export class FilterBuilderClass extends BaseFilterBuilderClass {
   }
 }
 
+/**
+ *  We want to support more complex negation than allowed by the normal filter tree, so we store negated in a property.
+ *  However, since the write function in cql.tsx doesn't know about this, at some point we need to convert it back to this class where negation exists as a type instead of property.
+ *  See the uncollapseNots function in cql.tsx for where this is done.
+ */
 export class CQLStandardFilterBuilderClass extends BaseFilterBuilderClass {
```

Comment:
For my understanding, could you break down what's going on here a little more. 
What does negation as a type vs property mean?

Reply:
yep very helpful, ty

Reply:
so at some point we turn what the UI has as a field into a group (which can be negated). 

So for clarity translating to something like this isn't possible/ undesired behaviour
does `-field_txt:"value"`